js/pages/samples.js

```diff
@@ -432,42 +522,471 @@ export const renderSamplesPage = async () => {
+
+        template += translateHTML(`<div class="row gy-3"  id="sampleInventory">
+            <div class="col-lg-2 col-xl-3"></div>
+            <div class="col-lg-8 col-xl-6">
+                <div class="consentHeadersFont" style="color:#606060;width:100%" data-i18n="samples.sampleInventory"></div>
+            </div>
+            <div class="col-lg-2 col-xl-3"></div>
+            ${samplesBody}
+        </div>
+        `);
 
         document.getElementById('root').innerHTML = template;
+
+        renderRequestAKitDisplay(participant);
+        bindEvents(participant);
+    });
+}
+
+const getParticipantMailToAddress = (participant) => {
+    const {
+        physicalAddress1, physicalAddress2, physicalCity, physicalState, physicalZip,
+        address1, address2, city, state, zip, isPOBox, yes
+    } = conceptId;
+    let invalidAddress = false;
+    let addressType;
+
+    const poBoxRegex = /^(?:P\.?\s*O\.?\s*(?:Box|B\.?)?|Post\s+Office\s+(?:Box|B\.?)?)\s*(\s*#?\s*\d*)((?:\s+(.+))?$)$/i;
+    const physicalAddressVal = participant?.[conceptId.physicalAddress1];
+
+    let addressObj;
+
+    if(physicalAddressVal && !poBoxRegex.test(physicalAddressVal)) {
+        addressObj = {
+            address_1: participant[physicalAddress1],
+            address_2: participant[physicalAddress2] || '',
+            city: participant[physicalCity],
+            state: participant[physicalState],
+            zip_code: participant[physicalZip], 
+        };
+        addressType = 'physical';
+    } else {
+        const addressLineOne = participant?.[address1];
+        const isPOBoxMatch = poBoxRegex.test(addressLineOne) || participant?.[isPOBox] === yes;
+        if(isPOBoxMatch) {
+            invalidAddress = true;
+        }
+        addressType = 'mailing';
+        addressObj = {
+            address_1: participant[address1],
+            address_2: participant[address2] || '',
+            city: participant[city],
+            state: participant[state],
+            zip_code: participant[zip]
+        };
+    }
+
+    let kitMailToAddress = `${addressObj.address_1 || ''}<br />
+            ${addressObj.address_2 ? `${addressObj.address_2}<br />` : ''}
+        ${addressObj.city || ''}${addressObj.state ? ',':''} ${addressObj.state || ''} ${addressObj.zip || ''}`;
+
+    return {
+        invalidAddress,
+        kitMailToAddress,
+        addressType,
+        addressObj
+    }
+}
+
+const renderRequestAKitDisplay = (participant) => {
+    const requestAKitInner = document.getElementById('requestAKitInner');
+    if(!requestAKitInner || participant[conceptId.kitRequestEligible] !== conceptId.yes) {
+        return;
+    }
+
+    const {collectionDetails, baseline, bioKitMouthwash, kitStatus, shippedDateTime, uniqueKitID, kitStatusValues: {initialized, assigned, addressPrinted, shipped}} = conceptId;
+
+    if ([initialized, addressPrinted, assigned].indexOf(participant[collectionDetails]?.[baseline]?.[bioKitMouthwash]?.[kitStatus]) > -1) {
```

Comment:
Optional: We could use `includes(participant[collectionDetails]?.[baseline]?.[bioKitMouthwash]?.[kitStatus])` here instead of `indexOf(participant[collectionDetails]?.[baseline]?.[bioKitMouthwash]?.[kitStatus]) > -1` for readability.